core/providers/mistral/mistral.go

```diff
@@ -237,12 +241,299 @@ func (provider *MistralProvider) SpeechStream(ctx context.Context, postHookRunne
+					Message: schemas.ErrRequestCancelled,
+					Error:   err,
+				},
+			}
+		}
+		if errors.Is(err, fasthttp.ErrTimeout) || errors.Is(err, context.DeadlineExceeded) {
+			return nil, providerUtils.NewBifrostOperationError(schemas.ErrProviderRequestTimedOut, err, providerName)
+		}
+		return nil, providerUtils.NewBifrostOperationError(schemas.ErrProviderDoRequest, err, providerName)
+	}
+
+	// Check for HTTP errors
+	if resp.StatusCode() != fasthttp.StatusOK {
+		defer providerUtils.ReleaseStreamingResponse(resp)
+		provider.logger.Debug(fmt.Sprintf("error from %s provider: %s", providerName, string(resp.Body())))
+		return nil, openai.ParseOpenAIError(resp, schemas.TranscriptionStreamRequest, providerName, request.Model)
+	}
+
+	// Create response channel
+	responseChan := make(chan *schemas.BifrostStream, schemas.DefaultStreamBufferSize)
+
+	// Start streaming in a goroutine
+	go func() {
+		defer close(responseChan)
+		defer providerUtils.ReleaseStreamingResponse(resp)
+
+		scanner := bufio.NewScanner(resp.BodyStream())
+		// Increase buffer size to handle large chunks
+		buf := make([]byte, 0, 64*1024) // 64KB initial buffer
+		scanner.Buffer(buf, 1024*1024)  // Allow up to 1MB tokens
+		chunkIndex := -1
+
+		startTime := time.Now()
+		lastChunkTime := startTime
+
+		var currentEvent string
+		var currentData string
+
+		for scanner.Scan() {
+			// Check if context is done before processing
+			select {
+			case <-ctx.Done():
+				return
+			default:
+			}
+
+			line := scanner.Text()
+
+			// Skip empty lines (event delimiter)
+			if line == "" {
+				// Process accumulated event if we have both event and data
+				if currentEvent != "" && currentData != "" {
+					chunkIndex++
+					provider.processStreamEvent(ctx, postHookRunner, currentEvent, currentData, request.Model, providerName, chunkIndex, startTime, &lastChunkTime, responseChan)
+				}
+				// Reset for next event
+				currentEvent = ""
+				currentData = ""
+				continue
+			}
+
+			// Parse SSE format
+			if strings.HasPrefix(line, "event: ") {
+				currentEvent = strings.TrimPrefix(line, "event: ")
+			} else if strings.HasPrefix(line, "data: ") {
+				currentData = strings.TrimPrefix(line, "data: ")
+			}
+		}
+
+		// Process any remaining event
+		if currentEvent != "" && currentData != "" {
+			chunkIndex++
+			provider.processStreamEvent(ctx, postHookRunner, currentEvent, currentData, request.Model, providerName, chunkIndex, startTime, &lastChunkTime, responseChan)
+		}
+
+		// Handle scanner errors
+		if err := scanner.Err(); err != nil {
+			provider.logger.Warn(fmt.Sprintf("Error reading stream: %v", err))
+			providerUtils.ProcessAndSendError(ctx, postHookRunner, err, responseChan, schemas.TranscriptionStreamRequest, providerName, request.Model, provider.logger)
+		}
+	}()
+
+	return responseChan, nil
+}
+
+// processStreamEvent processes a single SSE event and sends it to the response channel.
+func (provider *MistralProvider) processStreamEvent(
```

Comment:
lets make the function name more descriptive - `processTranscriptionStreamEvent`